ir/irutils.cpp

```diff
@@ -176,7 +175,10 @@ const IR::Expression *getDefaultValue(const IR::Type *type, const Util::SourceIn
 
 std::vector<const Expression *> flattenStructExpression(const StructExpression *structExpr) {
     std::vector<const Expression *> exprList;
-    for (const auto *listElem : structExpr->components) {
+    const auto *structType = structExpr->type->checkedTo<IR::Type_StructLike>();
```

Comment:
This is a little problematic because the structType can be `Type_Name`. I am not sure about a better way to handle this, however. At least we throw a bug in this case, which is easy to debug. 

For P4Testgen, this is easy to fix once #4215 is approved and merged. We can just add a pass that converts every `Type_Name` of a struct expression into a `Type_Structlike`. We should do this anyway.